pkg/expressions/funcfile/stage.go

```diff
@@ -0,0 +1,40 @@
+package funcfile
+
+import (
+	"rare/pkg/expressions"
+	"rare/pkg/slicepool"
+)
+
+type lazySubContext struct {
+	args []expressions.KeyBuilderStage
+	sub  expressions.KeyBuilderContext
+}
+
+func (s *lazySubContext) GetMatch(idx int) string {
+	if idx < 0 || idx >= len(s.args) {
+		return ""
+	}
+	return s.args[idx](s.sub)
+}
+
+func (s *lazySubContext) GetKey(name string) string {
+	return s.sub.GetKey(name)
```

Comment:
If you get rid of the sub-context (bad practice anyway?) this code wouldn't need a pool